nixos/modules/system/boot/systemd/sysupdate.nix

```
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.systemd.sysupdate;
+
+  writeDefinition = name: transferDefinition: pkgs.writeText
+    "${name}.conf"
+    (lib.generators.toINI { } transferDefinition);
+
+  listOfDefinitions = lib.mapAttrsToList
+    writeDefinition
+    (lib.filterAttrs (k: _: !(lib.hasPrefix "_" k)) cfg.transfers);
+
+  definitionsDirectory = pkgs.runCommand "systemd-sysupdate-definitions" { } ''
```

Bikeshed but name the directory by the name of the manpage?
```suggestion
  definitionsDirectory = pkgs.runCommand "sysupdate.d" { } ''
```

---

Also I have a feeling this definitions stuff we keep writing over and over again. Do we want to abstract it for all systemd config stuff?

---

I think that's worth doing. However I don't have a good idea yet how to do it. There are subtle differences to e.g. the repart module. Can we do this in a separate PR?

---

Yes. separate PR is ok